### PR TITLE
fix: workaround for reading uint32 data

### PR DESCRIPTION
**Pull request by trivoldus28**
In the same manner as uint64 because torch doesn't like uint

### CODE REVIEW

**Review by supersergiy — APPROVED** [review]
lgtm

**Comment by trivoldus28** [comment]
Seems like I should add a unit test. @supersergiy should I continue pushing to this branch?

**Comment by supersergiy** [comment]
Yeah, just push whatever's missing, and you should be able to press the merge button once the CI is green. You could either push the test as a separate commit, or do `git commit --amend` followed by `git push --force` to amend the existing commit

**Comment by trivoldus28** [comment]
@supersergiy Sorry you approved it but I thought you merged in the branch lol. What about that case when you actually did but I wanted to change something, like either fixing a bug, add more comments or more tests?

**Comment by supersergiy** [comment]
If I actually merged it in?

The CI shouldn't let me merge in before the test coverage is met. Merging in requires both reviewer approval and CI to be green. If the CI was green and the PR was merged in, but you want to make more modifications, just open a new PR contributing whatever new changes you see fit

**Comment by trivoldus28** [comment]
Yea makes sense.

**Pull request merged**
